src/main/java/core/basesyntax/service/impl/operation/SupplyOperation.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax.service.impl.operation;
+
+import core.basesyntax.dao.FruitStorageDao;
+import java.util.Objects;
+
+public class SupplyOperation implements OperationHandler {
+    private final FruitStorageDao storageDao;
+
+    public SupplyOperation(FruitStorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
+
+    @Override
+    public void doOperation(String fruitName, Integer quantity) {
+        if (quantity < 0) {
+            throw new IllegalArgumentException(
+                    "Quantity cannot be negative for supply operation: " + quantity
+            );
+        }
+
+        Integer previousQuantity;
+        Integer currentQuantity = storageDao.getQuantity(fruitName);
+
+        previousQuantity = Objects.requireNonNullElse(currentQuantity, 0);
+
+        Integer newQuantity = previousQuantity + quantity;
+        storageDao.setQuantity(fruitName, newQuantity);
```

Comment:
Let's simplify it, in` storageDao.getQuantity()` you already check that fruit in the storage, so it's quantity must be 0 or more 
```suggestion
        int currentQuantity = storageDao.getQuantity(fruitName);
        storageDao.setQuantity(fruitName, currentQuantity + quantity);
```